rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -355,62 +354,64 @@ object Reduce {
       * @param neu
       * @return
       */
-    def eval(neu: New)(implicit env: Env[Par]): M[Unit] = {
+    private def eval(neu: New)(implicit env: Env[Par]): M[Unit] = {
       def alloc(level: Int): Env[Par] =
         (env /: (0 until level).toList) { (_env, _) =>
           val addr: Par = GPrivate()
           _env.put(addr)
         }
 
-      eval(neu.p.get)(alloc(neu.bindCount))
+      // TODO: should cost of `new` construct depend on the # of bindings?
```

Comment:
Yes, new should cost based on the # of bindings. It's one hash evaluation per 2 names, so it seems pretty fair.